timeinterval/repeating_interval.go

```diff
@@ -0,0 +1,164 @@
+package timeinterval
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// RepeatingInterval describes an interval with recurring events distributed evenly by a fixed duration.
+// The interval can be bounded by either:
+// a fixed startsAt and endsAt
+// or by a fixed startsAt with a fixed number of Repetitions from which the endsAt will be derived.
+// or by a fixed endsAt with a fixed number of Repetitions from which the startsAt will be derived.
+type RepeatingInterval struct {
+	Interval Interval
+	RepeatIn time.Duration
```

Comment:
```suggestion
	RepeatEvery time.Duration
```
maybe? At least that make it clearer to me.

Reply:
I was also back and forth on the naming there. I actually started with `RepeatEvery`. The reason I changed it was because I felt the word `Every` could be misleading if the number of repetitions is limited say max 5. :thinking: